internal/route/api/v1/repo/path.go

```diff
@@ -0,0 +1,198 @@
+package repo
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+
+	"github.com/gogs/git-module"
+	"gogs.io/gogs/internal/context"
+	"gogs.io/gogs/internal/db/errors"
+)
+
+type GitRepoContent struct {
+	Type            string `json:"type"`
+	Target          string `json:"target,omitempty"`
+	Size            int64  `json:"size"`
+	Name            string `json:"name"`
+	Path            string `json:"path"`
+	Sha             string `json:"sha"`
+	URL             string `json:"url"`
+	GitURL          string `json:"git_url"`
+	HTMLURL         string `json:"html_url"`
+	DownloadURL     string `json:"download_url"`
+	Links           Links  `json:"_links"`
+	Content         string `json:"content,omitempty"`
+	SubmoduleGitURL string `json:"submodule_git_url,omitempty"`
+	Encoding        string `json:"encoding,omitempty"`
+}
+
+type Links struct {
+	Git  string `json:"git"`
+	Self string `json:"self"`
+	HTML string `json:"html"`
+}
+
+const (
+	// :base-url/:username/:project/raw/:refs/:path
+	templateDownloadURL = "%s/%s/%s/raw/%s"
+	// :base-url/repos/:username/:project/contents/:path
+	templateSelfLink = "%s/repos/%s/%s/contents/%s"
+	// :baseurl/repos/:username/:project/git/trees/:sha
+	templateGitURLLink = "%s/repos/%s/%s/trees/%s"
+	// :baseurl/repos/:username/:project/tree/:sha
+	templateHTMLLLink = "%s/repos/%s/%s/tree/%s"
+)
```

Comment:
It feels less ideal to have package-level constants. I would suggest to hard-code this string literals inside the function body. And refactor later when appropriate.

Reply:
Sure, moved them into the funcs and have left a TODO.